[CELEBORN-822][DOC] Add quick start guide (#1745)
apache/incubator-celeborn: docs/README.md

  See the License for the specific language governing permissions and
  limitations under the License.
---
Quick Start
===

> This documentation gives a quick start guide for running Spark with Celeborn.

Reviewer: When a project belongs to ASF first appears in a artical, we should call it "Apache XYZ"

```suggestion
This documentation gives a quick start guide for running Apache Spark with Apache Celeborn(Incubating).
```

### Download Celeborn
Download the latest Celeborn binary from the [Downloading Page](https://celeborn.apache.org/download/).

> Decompress the binary into ```$CELEBORN_HOME```

Reviewer: Use single backtick for inline code

```suggestion
Decompress the binary into `$CELEBORN_HOME`
```

Reviewer: seems we should guide user to `export CELEBORN_HOME` first

> ```angular2html

Reviewer: maybe `shell` or `bash`
```suggestion
```shell
```

tar -C <DST_DIR> -zxvf apache-celeborn-<VERSION>-bin.tgz
```

## Configure Logging and Storage
#### Configure Logging
```angular2html
cd $CELEBORN_HOME/conf

> mv log4j2.xml.template log4j2.xml

Reviewer: I suppose `cp` is encouraged rahter `mv`

```
#### Configure Storage

> Configure the directory to store shuffle data, for example $CELEBORN_HOME/shuffle

Reviewer: ```suggestion
Configure the directory to store shuffle data, for example `$CELEBORN_HOME/shuffle`
```

echo "celeborn.worker.storage.dirs=$CELEBORN_HOME/shuffle" > celeborn-defaults.conf
```

## Start Celeborn Service
#### Start Master
```angular2html
cd $CELEBORN_HOME
./sbin/start-master.sh
```
You should see ```Master```'s ip:port in the log:
```angular2html
INFO [main] NettyRpcEnvFactory: Starting RPC Server [MasterSys] on 192.168.2.109:9097
```
#### Start Worker
Use the Master's IP and Port to start Worker:
```angular2html
cd $CELEBORN_HOME
./sbin/start-worker.sh celeborn://${Master IP}:${Master Port}
```
You should see the following message in Worker's log:
```angular2html
23/07/22 11:39:23,546 INFO [main] MasterClient: connect to master 192.168.2.109:9097.
23/07/22 11:39:23,673 INFO [main] Worker: Register worker successfully.
```
And also the following message in Master's log:
```angular2html
23/07/22 11:39:23,650 INFO [dispatcher-event-loop-9] Master: Registered worker
Host: 192.168.2.109
RpcPort: 57806
PushPort: 57807
FetchPort: 57809
ReplicatePort: 57808
SlotsUsed: 0
LastHeartbeat: 0
HeartbeatElapsedSeconds: xxx
Disks:
  DiskInfo0: xxx
UserResourceConsumption: empty
WorkerRef: null
```

## Start Spark with Celeborn
#### Copy Celeborn Client to Spark's jars

> Celeborn release binary contains clients for Spark2.x and Spark3.x, copy the corresponding client jar into Spark's

Reviewer 2: ```suggestion
Celeborn release binary contains clients for Spark 2.x and Spark 3.x, copy the corresponding client jar into Spark's
```

```jars/``` directory:
```angular2html

> cp $CELEBORN_HOME/spark/${Celeborn Client Jar} $SPARK_HOME/jars/

Reviewer: ```suggestion
cp $CELEBORN_HOME/spark/<Celeborn Client Jar> $SPARK_HOME/jars/
```

  See the License for the specific language governing permissions and
  limitations under the License.
---
Quick Start
===
This documentation gives a quick start guide for running Apache Spark with Apache Celeborn(Incubating).

### Download Celeborn
Download the latest Celeborn binary from the [Downloading Page](https://celeborn.apache.org/download/).
Decompress the binary and set `$CELEBORN_HOME`
```shell
tar -C <DST_DIR> -zxvf apache-celeborn-<VERSION>-bin.tgz
export $CELEBORN_HOME=${Decompressed path}
```

## Configure Logging and Storage
#### Configure Logging
```shell
cd $CELEBORN_HOME/conf
cp log4j2.xml.template log4j2.xml
```
#### Configure Storage
Configure the directory to store shuffle data, for example `$CELEBORN_HOME/shuffle`
```shell
cd $CELEBORN_HOME/conf
echo "celeborn.worker.storage.dirs=$CELEBORN_HOME/shuffle" > celeborn-defaults.conf
```

## Start Celeborn Service
#### Start Master
```shell
cd $CELEBORN_HOME
./sbin/start-master.sh
```

> You should see ```Master```'s ip:port in the log:

Reviewer: ```suggestion
You should see `Master`'s ip:port in the log:
```

```shell
INFO [main] NettyRpcEnvFactory: Starting RPC Server [MasterSys] on 192.168.2.109:9097
```
#### Start Worker
Use the Master's IP and Port to start Worker:
```shell
cd $CELEBORN_HOME
./sbin/start-worker.sh celeborn://${Master IP}:${Master Port}
```
You should see the following message in Worker's log:
```shell
23/07/22 11:39:23,546 INFO [main] MasterClient: connect to master 192.168.2.109:9097.
23/07/22 11:39:23,673 INFO [main] Worker: Register worker successfully.
```
And also the following message in Master's log:
```shell
23/07/22 11:39:23,650 INFO [dispatcher-event-loop-9] Master: Registered worker
Host: 192.168.2.109
RpcPort: 57806
PushPort: 57807
FetchPort: 57809
ReplicatePort: 57808
SlotsUsed: 0
LastHeartbeat: 0
HeartbeatElapsedSeconds: xxx
Disks:
  DiskInfo0: xxx
UserResourceConsumption: empty
WorkerRef: null
```

## Start Spark with Celeborn
#### Copy Celeborn Client to Spark's jars
Celeborn release binary contains clients for Spark 2.x and Spark 3.x, copy the corresponding client jar into Spark's
```jars/``` directory:
```shell
cp $CELEBORN_HOME/spark/<Celeborn Client Jar> $SPARK_HOME/jars/
```
#### Start spark-shell

> Set ```spark.shuffle.manager``` to Celeborn's ShuffleManager, and turn off ```spark.shuffle.service.enabled```:

Reviewer: ```suggestion
Set `spark.shuffle.manager` to Celeborn's ShuffleManager, and turn off `spark.shuffle.service.enabled`:
```
